src/main/java/uk/gov/ons/ctp/response/action/service/CaseNotificationService.java

```diff
@@ -18,108 +17,134 @@
 import uk.gov.ons.ctp.response.collection.exercise.representation.CollectionExerciseDTO;
 
 /**
- * Save to Action.Case table for case creation life cycle events, delete for case close life cycle
- * events.
+ * Service for receiving notifications from case service Creates/Updates/Deletes cases in the
+ * action.case table
  */
 @Service
 public class CaseNotificationService {
   private static final Logger log = LoggerFactory.getLogger(CaseNotificationService.class);
 
   private static final int TRANSACTION_TIMEOUT = 30;
 
-  @Autowired private ActionCaseRepository actionCaseRepo;
+  private ActionCaseRepository actionCaseRepo;
 
-  @Autowired private ActionPlanRepository actionPlanRepo;
+  private ActionPlanRepository actionPlanRepo;
 
-  @Autowired private ActionService actionService;
+  private ActionService actionService;
 
-  @Autowired private CollectionExerciseClientService collectionSvcClientServiceImpl;
+  private CollectionExerciseClientService collectionSvcClientServiceImpl;
+
+  public CaseNotificationService(
+      ActionCaseRepository actionCaseRepo,
+      ActionPlanRepository actionPlanRepo,
+      ActionService actionService,
+      CollectionExerciseClientService collectionSvcClientServiceImpl) {
+    this.actionCaseRepo = actionCaseRepo;
+    this.actionPlanRepo = actionPlanRepo;
+    this.actionService = actionService;
+    this.collectionSvcClientServiceImpl = collectionSvcClientServiceImpl;
+  }
 
   @Transactional(
       propagation = Propagation.REQUIRED,
       readOnly = false,
       timeout = TRANSACTION_TIMEOUT)
   public void acceptNotification(final CaseNotification notification) throws CTPException {
-    final String actionPlanIdStr = notification.getActionPlanId();
-    final UUID actionPlanId = UUID.fromString(actionPlanIdStr);
-    final ActionPlan actionPlan = actionPlanRepo.findById(actionPlanId);
-    final UUID caseId = UUID.fromString(notification.getCaseId());
-    final UUID collectionExerciseId = UUID.fromString(notification.getExerciseId());
-    UUID partyId = null;
-    if (notification.getPartyId() != null) {
-      partyId = UUID.fromString(notification.getPartyId());
+
+    UUID caseId = UUID.fromString(notification.getCaseId());
+
+    switch (notification.getNotificationType()) {
+      case REPLACED:
+      case ACTIVATED:
+        ActionCase actionCase = createActionCase(notification);
+        saveActionCase(actionCase);
+        break;
+
+      case ACTIONPLAN_CHANGED:
+        updateActionCase(caseId, UUID.fromString(notification.getActionPlanId()));
+        break;
+
+      case DISABLED:
+      case DEACTIVATED:
+        actionService.cancelActions(caseId);
+        deleteActionCase(caseId);
+        break;
+
+      default:
+        log.with("notification_type", notification.getNotificationType())
```

Comment:
This should probably be a `throw new IllegalArgumentException("Invalid notification type")`